proposals/html-module-spec-changes.md

```diff
@@ -52,65 +71,89 @@ Questions/corrections/feedback are welcome!  I've left TODOs in several places w
+  - 1\.	Run the HTML5 parser on *source* to obtain the result *document*.
+  - 2\. Set *htmlModuleScript*[[document]] to *document*
+  - 2\. Let *scriptEntries* be an empty list of ScriptEntry Records (see definition in ES6 changes above).
+  - 3\.	For each HTMLScriptElement *script* in *document*:
+    - a\.	Let *se* be a new ScriptEntry record.
     - b\.	If *script* is inline:
-      - i\.	Set *se*[[IsInline]] = true
-      - ii\.	Set *se*[[RequestedModule]] = *script’s* Source Text Module Record
-      - iii\.	Set *se*[[SourceName]] = null
+      - i\.	Set *se*[[InlineModuleRecord]] = *script’s* Source Text Module Record
+      - ii\.	Set *se*[[ExternalScriptURL]] = null
     - c\.	Else  
-      - i\.	Set *se*[[IsInline]] = false
-      - ii\.	Set *se*[[RequestedModule]] = null
-      - iii\.	Set *se*[[SourceName]] = *script’s* src URL
-    - d\.	Append *se* to *record*.[[RequestedModules]]
-  - 5\.	Return *record*.
-- Change fetch a [single module script](https://html.spec.whatwg.org/#fetch-a-single-module-script) as follows:
-  - In step 9, allow HTML MIME type through in addition to JavaScript.
-  - In step 11, don’t unconditionally [create a module script](https://html.spec.whatwg.org/#fetching-scripts:creating-a-module-script).  Instead, key off the MIME type extracted in step 9, creating an HTML Module instead if we have an HTML MIME type.
-- Change step 5 of [fetch the descendants of a module script](https://html.spec.whatwg.org/#fetch-the-descendants-of-a-module-script) such that when record is an HTML Module, use [[SourceName]] from each *record*.[[RequestedModules]] instead of the Source Text Module Record field *record*.[[RequestedModules]].  This change basically just accounts for the differences in how Source Text Module Record and HTML Module Record store their descendant module URLs.
+      - i\.	Set *se*[[InlineModuleRecord]] = null
+      - ii\. Set *se*[[ExternalScriptURL]] = *script’s* src URL
+    - d\.	Append *se* to *scriptEntries*.
+  - 4\. Return a new HTML Module Record { [[Realm]]: *realm*, [[Environment]]: *undefined*, [[Namespace]]: *undefined*, [[Status]]: `"uninstantiated"`, [[EvaluationError]]: *undefined*, [[HostDefined]]: *htmlModuleScript*, [[RequestedModules]]: *scriptEntries*, [[DFSIndex]]: *undefined*, [[DFSAncestorIndex]]: *undefined* }.
+- Provide an implementation of the abstract operation HostGetDefaultExport(*module*) as the following:
+  1. Assert: *module* is an HTML Module.  Note: It is anticipated that in the future this may be expanded to support other module types (JSON etc).
+  1. Let *htmlModuleScript* be *module*.[[HostDefined]].
+  1. Let *document* be *htmlModuleScript*'s *document*.
+  1. Return *document*.
+- Change [fetch a single module script](https://html.spec.whatwg.org/#fetch-a-single-module-script) as follows:
+  - In step 9, allow `text/html` type through in addition to JavaScript.
```

Comment:
As per #742 I object to this.

Reply:
Yeah, that's fair.  I changed the language to reflect that the MIME type discussion is still ongoing, and linked [#742](https://github.com/w3c/webcomponents/issues/742).